app/assets/javascripts/initializers/runtime.js

```diff
@@ -7,6 +7,62 @@
  * if (Runtime.isNativeAndroid('podcastMessage')) { ... }
  */
 class Runtime {
+  /**
+   * This function returns a string combining the current Medium and OS
+   * that represents the current Context where the app is running.
+   *
+   * @returns {String} "Medium-OS", for example "Browser-Android"
+   */
+  static currentContext() {
+    return `${Runtime.currentMedium()}-${Runtime.currentOS()}`;
+  }
+
+  /**
+   * This function returns a string that represents the current Medium where
+   * the app is currently running. The currently supported mediums are Browser,
+   * ForemWebView and PWA.
+   *
+   * @returns {String} One of the supported Mediums or 'Unsupported'
+   */
+  static currentMedium() {
+    const pwaButtons = document.getElementById('pwa-nav-buttons');
+    if (/ForemWebView/i.test(navigator.userAgent)) {
+      return 'ForemWebView';
+    } else if (pwaButtons.classList.contains('pwa-nav-buttons--showing')) {
+      return 'PWA';
+    } else {
+      return 'Browser';
+    }
+  }
+
+  /**
+   * This function returns a string that represents the current OS where the app
+   * is currently running. The currently supported Operating Systems are
+   * Windows, Linux, macOS, Android and iOS.
+   *
+   * @returns {String} One of the supported Operating Systems or 'Unsupported'
+   */
+  static currentOS() {
+    const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
```

Comment:
Does this include Apple M1?

Reply:
Funny enough [it does](https://forem-team.slack.com/archives/C7GE84DEJ/p1616596843134900?thread_ts=1616596026.134100&cid=C7GE84DEJ) 😄 

I added a test case for this, so if it ever changes we can update accordingly also in the test suite.